src/App.jsx

```diff
@@ -1,3 +1,211 @@
+                  : "Show Liked Thoughts"}
+              </span>{" "}
+              ❤️️
+            </button>
+
+            <p>Total unique liked thoughts: {uniqueLikedCount}</p>
+            {showLikedThoughts && (
+              <ul>
+                {thoughts
+                  .filter((thought) => likedThoughtIds.includes(thought._id))
+                  .map((likedThought, index) => (
+                    <li key={index} className="liked-thought">
+                      {likedThought.message}
+                    </li>
+                  ))}
+              </ul>
+            )}
+          </div>
+          <form onSubmit={handleFormSubmit}>
+            <label htmlFor="newThoughtInput" className="text-label">
+              What&apos;s making you happy right now?
+            </label>
+            <input
+              type="text"
+              id="newThoughtInput"
+              name="newThought"
+              value={newThought}
+              onChange={(e) => {
+                const inputValue = e.target.value
+                setNewThought(inputValue)
+                setCharCount(inputValue.length)
+                setError("")
+              }}
+              placeholder="..."
+              className="input-field"
+            />
+            <p
+              className={charCount > 140 ? "char-count exceeded" : "char-count"}
+            >
+              {charCount}/140 characters
+            </p>
+
+            {error && <p style={{ color: "#FFADAD" }}>{error}</p>}
+            <button type="submit" className="submit-button">
+              ❤️{" "}
+              <span
+                style={{
+                  fontFamily: "'Fira Mono', monospace",
+                  fontWeight: "500",
+                }}
+              >
+                Send Happy Thought
+              </span>{" "}
+              ❤️️
+            </button>
+          </form>
+          <ul>
+            {thoughts.map((thought, index) => (
+              <li
+                key={index}
+                className={`thought-item ${isAddingThought ? "adding" : ""}`}
+              >
+                <div className="thought-content">
+                  <span className="message">{thought.message}</span>
+                </div>
+                <div className="thought-actions">
+                  <div className="left-actions">
+                    <button
+                      className={`like-button ${
+                        likedThoughtIds.includes(thought._id) && "liked"
+                      }`}
+                      onClick={() => likeThought(thought._id)}
+                    >❤️️ 
+                    </button>
+                    <span>x {thought.hearts}</span>
+                  </div>
+                  <div className="right-actions">
+                    <span>{timeAgo(thought.createdAt)}</span>
+                  </div>
+                </div>
+              </li>
+            ))}
+          </ul>
+        </>
+      )}
+    </div>
+  )
+}
```

Comment:
Your code looks very good and clean and is easy to read. You could divide the code into more components to make it a bit more easy to find what part you are looking at, but very nice work:)